agent/consul/leader.go

```diff
@@ -455,14 +455,6 @@ func (s *Server) initializeACLs(ctx context.Context) error {
 			}
 		}
 
-		// Check for configured management token from HCP. It MUST NOT override the user-provided initial management token.
-		if hcpManagement := s.config.Cloud.ManagementToken; len(hcpManagement) > 0 {
```

Comment:
See my comment below but I don't think we can get rid of this. HCP managed clusters rely on this to bootstrap ACLs which must occur before the following `insertAnonymousToken` call.